osh/bool_parse.py

```diff
@@ -48,7 +49,7 @@ class BoolParser(object):
   """Parses [[ at compile time and [ at runtime."""
 
   def __init__(self, w_parser):
-    # type: (WordParser) -> None
+    # type: (Union[WordParser, _StringWordEmitter]) -> None
```

Comment:
FYI: I noticed this Union issue too... it will require change when we translate (since it's not translatable) but of course the first step is to get it to pass MyPy in any way!